src/modules/musig/tests_impl.h

```diff
@@ -950,6 +987,145 @@ void musig_tweak_test(secp256k1_scratch_space *scratch) {
+        { /* 0 */
+            0xEA, 0x06, 0x7B, 0x01, 0x67, 0x24, 0x5A, 0x6F,
+            0xED, 0xB1, 0xB1, 0x22, 0xBB, 0x03, 0xAB, 0x7E,
+            0x5D, 0x48, 0x6C, 0x81, 0x83, 0x42, 0xE0, 0xE9,
+            0xB6, 0x41, 0x79, 0xAD, 0x32, 0x8D, 0x9D, 0x19,
+        },
+        { /* 1 */
+            0x14, 0xE1, 0xF8, 0x3E, 0x9E, 0x25, 0x60, 0xFB,
+            0x2A, 0x6C, 0x04, 0x24, 0x55, 0x6C, 0x86, 0x8D,
+            0x9F, 0xB4, 0x63, 0x35, 0xD4, 0xF7, 0x8D, 0x22,
+            0x7D, 0x5D, 0x1D, 0x3C, 0x89, 0x90, 0x6F, 0x1E,
+        },
+        { /* 2 */
+            0x70, 0x28, 0x8D, 0xF2, 0xB7, 0x60, 0x3D, 0xBE,
+            0xA0, 0xC7, 0xB7, 0x41, 0xDD, 0xAA, 0xB9, 0x46,
+            0x81, 0x14, 0x4E, 0x0B, 0x19, 0x08, 0x6C, 0x69,
+            0xB2, 0x34, 0x89, 0xE4, 0xF5, 0xB7, 0x01, 0x9A,
+        },
+        { /* 3 */
+            0x93, 0xEE, 0xD8, 0x24, 0xF2, 0x3C, 0x5A, 0xE1,
+            0xC1, 0x05, 0xE7, 0x31, 0x09, 0x97, 0x3F, 0xCD,
+            0x4A, 0xE3, 0x3A, 0x9F, 0xA0, 0x2F, 0x0A, 0xC8,
+            0x5A, 0x3E, 0x55, 0x89, 0x07, 0x53, 0xB0, 0x67,
+        },
+    };
+
+    for (i = 0; i < sizeof(combined_pk_expected)/sizeof(combined_pk_expected[0]); i++) {
+        size_t n_pks;
+        int has_second_pk;
+        int second_pk_idx;
+        switch (i) {
+            case 0:
+                /* [X1, X2, X3] */
+                n_pks = 3;
+                memcpy(pk_ser_tmp[0], pk_ser[0], sizeof(pk_ser_tmp[0]));
+                memcpy(pk_ser_tmp[1], pk_ser[1], sizeof(pk_ser_tmp[1]));
+                memcpy(pk_ser_tmp[2], pk_ser[2], sizeof(pk_ser_tmp[2]));
+                has_second_pk = 1;
+                second_pk_idx = 1;
+                break;
+            case 1:
+                /* [X3, X2, X1] */
+                n_pks = 3;
+                memcpy(pk_ser_tmp[2], pk_ser[0], sizeof(pk_ser_tmp[0]));
+                memcpy(pk_ser_tmp[1], pk_ser[1], sizeof(pk_ser_tmp[1]));
+                memcpy(pk_ser_tmp[0], pk_ser[2], sizeof(pk_ser_tmp[2]));
+                has_second_pk = 1;
+                second_pk_idx = 1;
+                break;
+            case 2:
+                /* [X1, X1, X1] */
+                n_pks = 3;
+                memcpy(pk_ser_tmp[0], pk_ser[0], sizeof(pk_ser_tmp[0]));
+                memcpy(pk_ser_tmp[1], pk_ser[0], sizeof(pk_ser_tmp[1]));
+                memcpy(pk_ser_tmp[2], pk_ser[0], sizeof(pk_ser_tmp[2]));
+                has_second_pk = 0;
+                second_pk_idx = 0; /* unchecked */
+                break;
+            case 3:
+                /* [X1, X1, X2, X2] */
+                n_pks = 4;
+                memcpy(pk_ser_tmp[0], pk_ser[0], sizeof(pk_ser_tmp[0]));
+                memcpy(pk_ser_tmp[1], pk_ser[0], sizeof(pk_ser_tmp[1]));
+                memcpy(pk_ser_tmp[2], pk_ser[1], sizeof(pk_ser_tmp[2]));
+                memcpy(pk_ser_tmp[3], pk_ser[1], sizeof(pk_ser_tmp[3]));
+                has_second_pk = 1;
+                second_pk_idx = 3;
```

Comment:
this should be 2?

I don't think this is the issue but my (toy code)[https://github.com/LLFourn/secp256kfun/pull/67] is failing on this case (while the other 3 are ok)

Reply:
`second_pk_idx = 2` and `second_pk_idx = 3` are both correct since they both get keyagg coefficient = 1. But agree that it's better to use the first one encountered here (and perhaps add comment that second_pk_idx = 3 is equally valid).